src/chat.rs

```diff
@@ -4231,6 +4232,74 @@ mod tests {
         assert_eq!(chatlist_len(&t, DC_GCL_ARCHIVED_ONLY).await, 1);
     }
 
+    #[async_std::test]
+    async fn test_unarchive_if_muted() -> Result<()> {
+        let t = TestContext::new_alice().await;
+
+        async fn msg_from_bob(t: &TestContext, num: u32) -> Result<()> {
+            dc_receive_imf(
+                t,
+                format!(
+                    "From: bob@example.org\n\
```

Comment:
Bob is `bob@example.net` and Alice is `alice@example.org` according to test-data public key metadata. It's better to use these addresses in tests against `new_alice` and `new_bob` accounts so they probably recognize self addresses.

Also not sure if `From: bob@example.net` should be `From: <bob@example.net>` instead. Definitely correct variant is `From: Bob <bob@example.net>`.

Reply:
yeah, i am always a bit confused about this example.net and example.org :)

for  `From: bob@example.net` (without square brackets), this regularly happens in the wild, so we should be able to deal with that. but i am also pretty sure that it is correct from some rfc.